stack-shell.nix

```diff
@@ -0,0 +1,15 @@
+# This nix expression is used by stack for setting up the build environment.
+# It referenced in `stack.yaml` so that stack uses it by default.
+{ }:
```

Comment:
Nit: this line can be removed. Stack probably uses some sort of `callPackage`, which is compatible with both functions taking dependencies as arguments, and a derivation. Since we use our own nixpkgs pin (as opposed to using the one from the nix channel), we're going for the latter option.

Reply:
Indeed, TIL.